apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/BuiltinAdapters.kt

```diff
@@ -0,0 +1,113 @@
+package com.apollographql.apollo.api
+
+import com.apollographql.apollo.api.internal.json.JsonWriter
+import com.apollographql.apollo.api.internal.json.Utils
+import com.apollographql.apollo.api.internal.json.use
+import okio.Buffer
+
+/**
+ * Builtin CustomScalarTypeAdapter provided for convenience. Encoding is most of the times straightforward. Decoding
+ * can involve coercion. If you need stricter decoding or different logic, define your own adapter
+ */
+object BuiltinScalarTypeAdapters {
+  val STRING_ADAPTER = adapterWithDefaultEncode { jsonElement ->
+    when (jsonElement) {
+      is JsonElement.JsonString -> jsonElement.value
+      is JsonElement.JsonNull -> null
+      is JsonElement.JsonBoolean -> jsonElement.value.toString()
+      is JsonElement.JsonNumber -> jsonElement.value.toString()
+      is JsonElement.JsonObject,
+      is JsonElement.JsonList -> {
```

Comment:
`JsonElement` is the new, more type safe `CustomTypeValue`. It makes it clear that `CustomTypeAdapters` ultimately work on Json:

```kotlin
interface CustomScalarTypeAdapter<T> {
  fun decode(jsonElement: JsonElement): T
  fun encode(value: T): JsonElement
}
```

I know GraphQL is not Json-only but so far it is so we'll see when (if...) that changes.